### DISCUSSION TITLE
Custom epoch num AND L-R mirroring off

### DISCUSSION THREAD

**Discussion by zalyalovi-arch** [MIC-DKFZ/nnUNet#2987] [Q&A]
Hi everyone.
I've been using nnunet for 2 months now and as for a non-IT guy (I'm a radiation oncologist actually) this thing is like a miracle.
I've got really excited seeing what the nnunet can do.
Now I'm working on CT segmentation project. I've got 100 cases with 31 labels and after a month of trying different ways to train the model I realized I need a custom trainer with left-right mirroring off and ~300 epochs long (I know I can use some default trainers like NoMirroring or onlyMirror01 for 1000 epochs and overfitting wouldn't be a problem, but I'm short on resources and have only ~40 hours of GPU time per week, with my epoch times it could take ~8 weeks to train a model). Now I stop training early when Dice scores stop to increase, but I guess it would be better to use nnunet as it was designed - to do full training.
So, the question is - is there a proper way to change learning rate to make a 300 epochs long trainer and turn off left-right mirroring at the same time? The best I can find is nnUNetTrainer_250epochs_NoMirroring (but I want to turn off only LR mirroring) and nnUNetTrainer_onlyMirror01 (it's 1000 epochs long).
I would appreciate any help.
Best.

**Comment by EmergentBehaviour** [comment]
You should be able to make a custom nnUNetTrainer class off of one of the existing 01 mirroring nnUNetTrainer variants like nnUNetTrainer_onlyMirror01_1500ep, and changing the number of epochs with self.num_epochs.

**Reply by EmergentBehaviour** [reply]
If this is your first time using your own modifications to a repo on the fly, just make sure to pip install -e . on your modified version of the code base so your system knows to include your new class.
